### ISSUE TITLE
:path in deps.edn for third-party drivers require absolute paths

### ISSUE THREAD

**Issue by luizarakaki** [metabase/metabase#22374]
**Describe the issue**
Currently, to build third-party drivers that live outside the Metabase core repo, it is necessary to define absolute paths in the `deps.edn` file. This is a problem for developers that have a CI process.

**Logs**
Exasol case example:
PR migrating Lein to tools: https://github.com/exasol/metabase-driver/pull/36/files

When building the driver, it fails with an error:
```
Command "clojure -X:dev:build" fails with this error:

Step "Copy :exasol source files" failed with error "Assert failed: All dirs should be absolute, got: [\"./src\" \"./resources\"]\n(every? u/absolute? dirs)"
```
Build link: https://github.com/exasol/metabase-driver/runs/6260237010?check_suite_focus=true

Assertion causing the error:
https://github.com/metabase/metabase/blob/master/bin/build-drivers/src/build_drivers/copy_source_files.clj#L11

**Expected behavior**
It should be possible to use relative paths in deps.end file key :paths

**Additional context**
Jeff Evans started to fix this issue in this PR #20279. The path fix looks done, this PR has a broader scope.

**Comment by flamber** [comment]
Also need to fix our reference driver https://github.com/metabase/sudoku-driver/issues/5

**Comment by dpsutton** [comment]
good catch @flamber i've already submitted a [PR](https://github.com/metabase/sudoku-driver/pull/6) to fix that and a [PR](https://github.com/metabase/sudoku-driver/pull/7) to fix the driver so others can use the demo.

**Comment by dpsutton** [comment]
This bug came about in https://github.com/metabase/metabase/pull/17608 . We used to allow relative paths. 

#### workaround:
we need to pass in an absolute path. Currently the `exec-args` is passing in a relative path (`"."`). We can add the current working directory when building the driver and it will use that.

```shell
clj -X:build :project-dir "\"$(pwd)\""
```
(the quoting gets annoying).

#### Background
I'm not 100% up-to-speed on our driver build code yet but what I think we are doing:
- When building the jars we do some surgery to remove anything that comes from metabase itself.
- we're compiling the namespaces to speed up load time
- I'm not exactly sure where relative vs absolute fits in here

But it seems like our build scripts are a bit annoying for 3rd party. It requires the 3rd party driver to sit alongside the metabase directory.

3rd party developers reference our build scripts like so:
```clojure
 ;; build the driver with clojure -X:build
 :aliases
 {:build
  {:extra-deps {metabase/metabase-core {:local/root "../metabase"}
                metabase/build-drivers {:local/root "../metabase/bin/build-drivers"}}
   :exec-fn    build-drivers.build-driver/build-driver!
   :exec-args  {:driver      :sudoku
                :project-dir "."
                :target-dir  "./target"}}}
```

I think in an ideal world they could reference it like so:

```clojure
{:deps {com.github.metabase/metabase {:git/tag "v0.43.0", :git/sha "ee686f"}
        com.github.metabase/build-drivers {:git/tag "v0.1.0" :git/sha "tbd"}}}
```

The difference here is that
- metabase does not have to be a sibling of the 3rd party driver. It can just depend upon it as a normal project
- the build scripts are a separate project that knows how to build a driver

**Comment by dragonsahead** [comment]
Is this still relevant due to our latest change @dpsutton ?
